econml/utilities.py

```diff
@@ -697,3 +790,518 @@ def predict(self, Xt):
+               np.array([_safe_norm_ppf(1-alpha/2, loc=p, scale=err)
+                 for p, err in zip(self.coef_, self.coef_stderr_)])
+
+    def intercept__interval(self, alpha=.05):
+        """
+        Parameters
+        ----------
+        alpha : float
+            The confidence level. Will calculate the alpha/2-quantile and the (1-alpha/2)-quantile
+            of the parameter distribution as confidence interval
+        Returns
+        -------
+        intercept__interval : {tuple ((p,) array, (p,) array), tuple (float, float)}
+            The lower and upper bounds of the confidence interval of the intercept(s)
+        """
+        if not self._fit_intercept:
+            return (0 if self._n_out==0 else np.zeros(self._n_out)), (0 if self._n_out==0 else np.zeros(self._n_out))
+
+        if self._n_out == 0:
+            return _safe_norm_ppf(alpha/2, loc=self.intercept_, scale=self.intercept_stderr_),\
+                    _safe_norm_ppf(1-alpha/2, loc=self.intercept_, scale=self.intercept_stderr_)
+        else:
+            return np.array([_safe_norm_ppf(alpha/2, loc=p, scale=err)
+                 for p, err in zip(self.intercept_, self.intercept_stderr_)]),\
+               np.array([_safe_norm_ppf(1-alpha/2, loc=p, scale=err)
+                 for p, err in zip(self.intercept_, self.intercept_stderr_)])
+
+    def predict_interval(self, X, alpha=.05):
+        """
+        Parameters
+        ----------
+        X : (n, d) array like
+            The covariates on which to predict
+        alpha : float
+            The confidence level. Will calculate the alpha/2-quantile and the (1-alpha/2)-quantile
+            of the parameter distribution as confidence interval
+        Returns
+        -------
+        prediction_intervals : {tuple ((n,) array, (n,) array), tuple ((n,p) array, (n,p) array)}
+            The lower and upper bounds of the confidence intervals of the predicted mean outcomes
+        """
+        return np.array([_safe_norm_ppf(alpha/2, loc=p, scale=err)
+                 for p, err in zip(self.predict(X), self.prediction_stderr(X))]),\
+               np.array([_safe_norm_ppf(1-alpha/2, loc=p, scale=err)
+                 for p, err in zip(self.predict(X), self.prediction_stderr(X))])
+
+
+class LassoCVWrapper:
+    """Helper class to wrap either LassoCV or MultiTaskLassoCV depending on the shape of the target."""
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+    def fit(self, X, Y):
+        assert shape(X)[0] == shape(Y)[0]
+        assert ndim(Y) <= 2
+        if ndim(Y) == 2 and shape(Y)[1] > 1:
+            self.model = MultiTaskLassoCV(*self.args, **self.kwargs)
+        else:
+            self.model = LassoCV(*self.args, **self.kwargs)
+        self.model.fit(X, Y)
+        return self
+
+    def predict(self, X):
+        return self.model.predict(X)
+
+class WeightedSplitter:
```

Comment:
Similarly, if we want to merge this in master we need to add unit tests.

Reply:
@kbattocchi we need to remove the `WeightedSplitter`. The `WeightedKFold` to be pushed in the PR as described in #103 is the only thing we need. This class is legacy from my branch

Reply:
Handled in next push.